### PR TITLE
JSSteam and JSSystem: move to Classmapper

### PR DESCRIPTION
**Pull request by verpeteren**
Please review
Note: needs a merge with Nidium/classmapper to work with 'CreateUniqueInstance'

### CODE REVIEW

File: src/Binding/JSStream.cpp [hunk 1]
```diff
@@ -112,7 +60,7 @@ void JSStream::onMessage(const Core::SharedMessages::Message &msg)
     JS::RootedValue onerror_callback(m_Cx);
     JS::RootedValue rval(m_Cx);
 
-    JS::RootedObject obj(m_Cx, m_JSObject);
+    JS::RootedObject obj(m_Cx, this->getJSObject());
```
Comment (paraboul) [inline src/Binding/JSStream.cpp]: `obj` can be replaced by `m_Instance`

File: src/Binding/JSSystem.cpp [hunk 2]
```diff
@@ -25,48 +25,21 @@
...
-    JS_FN("language", nidium_nidium_language, 0, NIDIUM_JS_FNPROPS),
-#endif
-#endif
-                                         JS_FS_END };
-// }}}
-
 // {{{ Implementation
-static bool
-nidium_system_getOpenFileStats(JSContext *cx, unsigned argc, JS::Value *vp)
+JSSystem *JSSystem::Constructor(JSContext *cx, JS::CallArgs &args,
```
Comment (paraboul) [inline src/Binding/JSSystem.cpp]: ::Constructor implementation can be removed now. (I recently default'ed it to ClassMapper).

File: src/Binding/JSSystem.cpp [hunk 3]
```diff
@@ -118,15 +91,35 @@ static bool nidium_nidium_language(JSContext *cx, unsigned argc, jsval *vp)
...
+    return funcs;
+}
+
 void JSSystem::RegisterObject(JSContext *cx)
 {
-    JS::RootedObject global(cx, JS::CurrentGlobalOrNull(cx));
-    JS::RootedObject systemObj(
-        cx, JS_DefineObject(cx, global, "System", &system_class, nullptr, 0));
-    JS_DefineFunctions(cx, systemObj, system_funcs);
+    JSSystem::ExposeClass<0>(cx, "System");
```
Comment (paraboul) [inline src/Binding/JSSystem.cpp]: The className needs to be different. You can set this to "NidiumSystem".
Also the <0> is not required

File: src/Binding/JSSystem.cpp [hunk 4]
```diff
@@ -119,7 +111,7 @@ JSFunctionSpec *JSSystem::ListMethods()
 void JSSystem::RegisterObject(JSContext *cx)
 {
     JSSystem::ExposeClass<0>(cx, "System");
-    JSSystem::CreateUniqueInstance(cx, new JSSystem(), "System");
+    JSSystem::CreateUniqueInstance(cx, new JSSystem(), "NidiumSystem");
```
Comment (paraboul) [inline src/Binding/JSSystem.cpp]: No it was on ExposeClass not CreateUniqueInstance.
Also the new doc is wrong then.

Comment (efyx) [inline src/Binding/JSSystem.cpp]: Why renaming it to NidumSystem ?

Comment (paraboul) [inline src/Binding/JSSystem.cpp]: It's the name of the prototype, note the object

File: docs/JSSystem.cpp.py [hunk 5]
```diff
@@ -26,8 +26,8 @@
                                         ("sockets", "Open sockes", "integer"),
                                         ("files", "Open Files", "integer")]))
 )
-FunctionDoc( "System.language", "Get the system language.",
-    SeesDocs( "System.language" ),
+FunctionDoc( "NidiumSystem.language", "Get the system language.",
```
Comment (paraboul) [inline docs/JSSystem.cpp.py]: No it's still System.
NidiumSystem is just the name of the class and should not be accessed directly.

**Pull request merged**